cmd/nvidia_gpu/nvidia_gpu.go

```diff
@@ -48,6 +49,10 @@ func main() {
 	mountPaths := []gpumanager.MountPath{
 		{HostPath: *hostPathPrefix, ContainerPath: *containerPathPrefix},
 		{HostPath: *hostVulkanICDPathPrefix, ContainerPath: *containerVulkanICDPathPrefix}}
+
+	nvmlManager := nvmlmanager.GetInstance()
```

Comment:
Why not consolidate nvml lifecycle management in the main function itself? That way we do not have to pass the nvmlmanager object around.  It will also prevent any object from incorrectly calling `nvmlmanager.Shutdown()` while some other object may still be using it.

What if we re-write this main function to make the steps more clear:
1.  Create a new method to check if drivers are installed, and keep looping till we detect drivers are installed.
2.  Call nvml.Init() and defer nvml.Shutdown().
3.  Start gpumanager
4.  Start metrics
5.  Start any new components

What do you think?

Reply:
yeah i guess that will be more clear and save us the trouble of introducing the manager. Will do it